src/Terrain/Maps/MapReader.java

```diff
@@ -0,0 +1,171 @@
+      { SE, GR, GR, CT, GR, RD, HQ, FC, SE, SE }, // 2
+      { SE, FR, FR, RD, RD, RD, GR, FC, SE, SE }, // 3
+      { SE, RD, RD, RD, GR, CT, GR, GR, MT, SE }, // 4
+      { SE, RD, FR, SE, SE, SE, GR, GR, CT, SE }, // 5
+      { SE, RD, SE, SE, SE, SE, GR, GR, MT, SE }, // 6
+      { SE, RD, GR, SE, SE, FR, FR, GR, FR, SE }, // 7
+      { SE, RD, RD, RD, RD, RD, FR, GR, CT, SE }, // 8
+      { SE, GR, CT, GR, CT, RD, GR, GR, GR, SE }, // 9
+      { SE, FR, GR, MT, MT, RD, SE, SE, RD, SE }, // 10
+      { SE, FC, GR, GR, CT, RD, SE, SE, GR, SE }, // 11
+      { SE, GR, HQ, RD, RD, RD, RD, GR, CT, SE }, // 12
+      { SE, FC, FC, FC, GR, CT, SE, GR, CT, SE }, // 13
+      { SE, SE, SE, SE, SE, SE, SE, SE, SE, SE } };// 14
+  private static XYCoord[] co1Props = { new XYCoord(1, 5), new XYCoord(1, 6), new XYCoord(2, 6), new XYCoord(2, 7),
+      new XYCoord(3, 7) };
+  private static XYCoord[] co2Props = { new XYCoord(11, 1), new XYCoord(12, 2), new XYCoord(13, 1), new XYCoord(13, 2),
+      new XYCoord(13, 3) };
+  private static XYCoord[][] properties = { co1Props, co2Props };
+  private static MapInfo info = new MapInfo(mapName, terrainData, properties);
+
+  /**
+   * Tells the MapReader to read in the maps.
+   */
+  public static ArrayList<MapInfo> readMapData()
+  {
+    ArrayList<MapInfo> importMaps = new ArrayList<MapInfo>();
+
+    try
+    {
+      final File folder = new File("res/map");
+
+      for( final File fileEntry : folder.listFiles() )
+      {
+        // we aren't checking subdirectories, yet
+        if( !fileEntry.isDirectory() )
+        {
+          try
+          {
+            // System.out.println();
+            String mapName = fileEntry.getName();
+            if( !mapName.contains(".map") )
+              continue;
+            mapName = mapName.replaceAll("_", " ");
+            mapName = mapName.replaceAll(".map", "");
+            System.out.println("Parsing map: " + mapName);
+            Scanner scanner = new Scanner(fileEntry);
+            String line = scanner.nextLine();
+            ArrayList<ArrayList<Environment.Terrains>> terrainData = new ArrayList<ArrayList<Environment.Terrains>>();
+
+            ArrayList<ArrayList<XYCoord>> properties = new ArrayList<ArrayList<XYCoord>>(4);
+            properties.add(new ArrayList<XYCoord>());
+            properties.add(new ArrayList<XYCoord>());
+            properties.add(new ArrayList<XYCoord>());
+            properties.add(new ArrayList<XYCoord>());
+
+            for( int i = 0; i < line.length() / 4; i++ )
+            {
+              terrainData.add(new ArrayList<Environment.Terrains>());
+            }
+            for( int yCoord = 0; scanner.hasNextLine(); line = scanner.nextLine() )
+            {
+              for( int i = 0; i < line.length(); i += 4 )
+              {
+                int sideNumber = 99;
+                try
+                {
+                  sideNumber = Integer.parseInt(("" + line.charAt(i) + line.charAt(i + 1)).trim());
```

Comment:
This kind of string addition is not very performant, since it generally requires copying the entire string contents as each piece is added (since strings are immutable). Not that it's too expensive here, but FYI. Can we just grab the substring we care about in one shot instead of kludging a char-to-string-to-int conversion?